src/app/components/fabric-post/fabric-post.component.ts

```diff
@@ -1,117 +1,211 @@
 import { Component, Inject } from '@angular/core';
 import { fabric } from 'fabric';
-import { POST_DEFAULT_BORDER, POST_DEFAULT_BORDER_THICKNESS } from 'src/app/utils/constants';
+import {
+  POST_DEFAULT_BORDER,
+  POST_DEFAULT_BORDER_THICKNESS,
+} from 'src/app/utils/constants';
 
-const AUTHOR_OFFSET = 65
-const DESC_OFFSET = 80
-const CONTENT_EXTRA_HEIGHT = 55
+const AUTHOR_OFFSET = 65;
+const DESC_OFFSET = 80;
+const CONTENT_EXTRA_HEIGHT = 55;
 
 @Component({
   selector: 'app-fabric-post',
   templateUrl: './fabric-post.component.html',
-  styleUrls: ['./fabric-post.component.scss']
+  styleUrls: ['./fabric-post.component.scss'],
 })
 export class FabricPostComponent extends fabric.Group {
+  constructor(@Inject(Object) options: any) {
+    var canvas = new fabric.Canvas('c');
+    // FROM : http://jsfiddle.net/illumine/Avvxn/
 
-  constructor(@Inject(Object) options:any) { 
-    var title = new fabric.Textbox(options.title, {
-      name: 'title',
-      width: 280,
-      left: 18,
-      top: 60,
-      fontSize: 18,
-      fontWeight: 'bold',
-      fontFamily: 'Helvetica',
-      fill: '#000000',
-      splitByGrapheme: true
-    });
-  
-    var author = new fabric.Textbox(options.author, {
-      name: 'author',
-      width: 300,
-      left: 18,
-      top: title.getScaledHeight() + AUTHOR_OFFSET,
-      fontSize: 13,
-      fontFamily: 'Helvetica',
-      fill: '#555555',
-      splitByGrapheme: true,
-    });
+    function wrapCanvasText(t, canvas, maxW, maxH) {
```

Comment:
Wrapper function looks good! Do you mind moving it into `utils/FabricUtils.ts`? That file is responsible for any helper functions associated with anything fabric-related. 

You can also remove the `canvas` parameter from this function as well since `_canvas` is already set in `utils/FabricUtils.ts`.